test/plugin/base_test.rb

```diff
@@ -544,6 +544,33 @@ def test_struct_payload_log
     end
   end
 
+  def test_struct_payload_malformat_log
+    setup_gce_metadata_stubs
+    message = 'test message'
+    setup_logging_stubs do
+      d = create_driver
+      d.emit(
+        'int_as_key' => { 1 => message },
```

Comment:
I'd just use `'int_key'`, `'array_key'`, `'hash_key'`, `'nil_key'`. Also, you forgot symbol (e.g., `{ :symbol => message }`).